resources/ts/Components/ButtonIcon.tsx

```diff
@@ -19,9 +20,11 @@ export const ButtonIcon = ({
 }: Props) => {
   return (
     <button
-      className={classNames(
-        "hover:bg-icon-gray-light focus:bg-icon-gray-dark group-focus:animate-pulse p-3 rounded-full grow-0 shrink-0 w-[2.8rem] h-[2.8rem] inline-flex items-center justify-center",
-        className
+      className={twMerge(
```

Comment:
I see that twMerge merges classes without conflicts.
Is it something we want to have by default?
Maybe we could integrate it in the `classNames` utility function?
Or is it overkill

Reply:
Until now we have been able to go without `twMerge` function just by using conditionals:

e.g.
Instead of 
`twMerge('mb-3 mb-6')`

We have just been doing
`condition ? 'mb-3' : 'mb-6'`

I'm not opposed to introducing `twMerge` if it is really needed. Then I would also integrate it in classNames.
But it is an extra runtime dependency we haven't needed until now.

Reply:
To me the main benefit of `twMerge` is for passing extra classes multiple levels down. I do agree for most use cases, I prefer a behavioral prop on the components that then affect their appearance. But for a lot of one off style changes it's not always realistic without making the components API too bloated, so I like the merging approach as you can easily pass a margin or border class multiple levels down and have it override only the necessary classes.

But I understand the will to limit the number of runtime dependencies, I'd say PPF is ok on the basis that it's still a mainly back-end project with not that much front-end for now so we have some leeway.